core/src/main/java/cucumber/runtime/RuntimeOptions.java

```diff
@@ -214,6 +222,10 @@ private void parse(List<String> args) {
             junitOptions.clear();
             junitOptions.addAll(parsedJunitOptions);
         }
+        if(orderTypeData.containsKey(OrderType.TYPE_NAME)) {
+        	orderType = OrderTypeFactory.getOrderType(orderTypeData);
+        	orderType.checkVariableValues(orderTypeData);
```

Comment:
The exact syntax of CLI commands is also specific to the runtime options parser. Checking if the input is valid should be done here.

Additionally any time when `--count` is used it's argument should be a number and should non-negative. Tying the validation to the order type makes this more complex then needed. 

I think it might be better to make `--count` and `--order` independent variables and simply ignore `--count` when not relevant.

Reply:
The `--count`  is used only for the `RandomOrderType` thus the check was specific to that type. That required the test for number etc to be done later.

Idea of making `--count` applicable for all types, to be used if required, does simplifies things. If so then the `orderTypeData` map is also redundant and can be removed. 

Later on if a future `OrderType` requires an additional variable from CLI it needs to be handled accordingly.